### PR TITLE
Remove server startup timeout

### PR DESCRIPTION
**Pull request by jakewins**
The startup block for the server is no longer wrapped in a timeout. This
startup timout has been severely watered down - upgrades and recovery happen
outside it, and HA has it disabled entirely. As such, it does not guard against
any known blocking startup actions, and remains only for historical reasons.

This commit removes it entirely, which simplifies the code.

We also remove all preflight tasks other than one. The upgrade and recovery
ones because they no longer need to happen outside the timeout, and the
config validating one becuase it was not used and is covered by the new Config
system usage in server instead

### CODE REVIEW

File: enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseNeoServer.java [hunk 1]
```diff
@@ -82,7 +78,8 @@ public EnterpriseNeoServer( ConfigurationBuilder configurator, Database.Factory
     protected PreFlightTasks createPreflightTasks()
     {
         return new PreFlightTasks( logProvider,
-                new EnsurePreparedForHttpLogging( configurator.configuration() ), new PerformUpgradeIfNecessary(
+                new EnsurePreparedForHttpLogging( configurator.configuration() ),
+                new PerformUpgradeIfNecessary(
                         getConfig(), configurator.getDatabaseTuningProperties(), logProvider,
                         StoreUpgrader.NO_MONITOR ), new PerformRecoveryIfNecessary( getConfig(),
                         configurator.getDatabaseTuningProperties(), logProvider ) );
```
Comment (chrisvest) [inline enterprise/server-enterprise/src/main/java/org/neo4j/server/enterprise/EnterpriseNeoServer.java]: Did you miss removing the upgrade preflight task from the EnterpriseNeoServer? You removed it from the CommunityNeoServer.

**Comment by chrisvest** [comment]
Nice to see this get removed. Perhaps the `EnsurePreparedForHttpLogging` code can be inlined into the normal start up process, and the whole preflight infrastructure can be removed.

**Comment by davidegrohmann** [comment]
@jakewins licenses missing and it needs rebase

**Comment by davidegrohmann** [comment]
@jakewins it looks like you forgot to remove one test: `StartupTimeoutFunctionalTest.shouldTimeoutIfStartupTakesLongerThanTimeout`

**Pull request merged**